### ISSUE TITLE
[Vote ended on 2022-06-03] What to do with a lack of collection inclusion reviews?

### ISSUE THREAD

**Issue by Andersson007** [ansible-community/community-topics#97]
##### Context

According to the [Steering Committee guidelines](https://docs.ansible.com/ansible/devel/community/steering/community_steering_committee.html), the Committee members review
collections [submitted](https://github.com/ansible-collections/ansible-inclusion/discussions) for inclusion in the Ansible community package.

The inclusion discussions are happening in the [ansible-inclusion](https://github.com/ansible-collections/ansible-inclusion/discussions) repository.

At the moment, a collection to be included in the package **MUST** be approved by, at least, **TWO** Committee members.

##### Problem to discuss and solve

- We have a lack of reviews, therefore, the submitted collection can't get into the package.

- We have a big inclusion backlog: new collections are submitted, old ones stay there.

##### How the process works now 

In a nutshell, the process now consists of the following steps:
1. A Steering Committee member copies the [Collection checklist](https://github.com/ansible-collections/overview/blob/main/collection_checklist.md) that reflects the [Community collection requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst) to a discussion.
2. The member goes through the checklist checking if the collection satisfies the requirements:
  - If it satisfies a condition, the member marks the corresponding checkbox complete.
  - If it does not satisfy, the member add a comment under the condition starting with `MUST FIX`.
  - If there are something that can be improved, the comment starts with `SHOULD FIX`.
3. After the collection maintainers fix all `MUST FIX` points and report about it:
  - The member checks them.
  - If there are no other concerns, the member explicitly approves the inclusion. 

##### Possible solutions to the lack of reviews problem

1. IMO, to fix the situation, the current workflow must be redesigned OR see option b below.
2. (Your solution here)

Possible implementation options:
a) Change the inclusion requirements: for a collection to be included it's enough to have 1 Steering Committee member review and 1 Community member review (instead of 2 Committee reviews as is now). The Community member must not be involved in the collection. This can potentially increase the number of volunteers but with no guarantee.
b) To expand the [Steering Committee Chairperson's responsibilities](https://docs.ansible.com/ansible/devel/community/steering/steering_committee_membership.html#chairperson) to give them power to assign Committee members to review the collection.
  - We create a list of members who have experience in collection/module development/maintenance.
  - Volunteers are welcome (@andersson007 will often be one of them). Before assigning, the chairperson (or another member) will call for volunteers.
  - If there are two volunteers, no other action is needed.
  - If there are one or no volunteers, the chairperson will assign the inclusion request trying to be fair (one-by-one) and considering a person's circumstances (e.g. busy time at day job, PTO, family reasons, etc).

c) (your option here)

Please share your thoughts, it's really a problem now.

**Comment by MarkusTeufelberger** [comment]
Some (a lot?) of these checks can be automated. It sounds very tedious to have 2 humans look and mark checkboxes if a `README.md` file exists for example.

**Comment by felixfontein** [comment]
That part's also related to #15. Some of the things are relatively easy to check automatically. Quite a few are not, though, and these are usually the more interesting things.

**Comment by Andersson007** [comment]
> Quite a few are not, though, and these are usually the more interesting things.

Yep, reviewing by at least two persons is imo important as this (i think almost always) happened that the second reviewer spotted things that the first one had missed (things that is hard to automate like adhering the dev conventions).
Automation would definitely be very helpful, though it won't fix the main issue.

**Comment by mariolenz** [comment]
I'm quite new to the steering committee, so I don't really feel qualified to discuss this. But one thing I can say:

> Yep, reviewing by at least two persons is imo important as this (i think almost always) happened that the second reviewer spotted things that the first one had missed

I fully agree. Especially since the first reviewer probably already had _lots_ of looks at the collection. In my experience, if someone didn't spot a problem at the first couple of times, they won't at the hundreth time, either. At least, that's what happens to me all of the time.

One other thing: I've had a look at [ansible-inclusion](https://github.com/ansible-collections/ansible-inclusion/discussions) and I don't find the categories really helpful. For example, under `New collection reviews` there are really _new_ inclusion requests that nobody had had at look at yet, but also some that already had dozens of comments. Maybe there should be more categories, like `First review needed`, `First review in progress`, `Second review needed` and `Second review in progress`. At the moment, I wouldn't even know where to start.

If this doesn't make sense, just ignore me. As I've already said, I'm quite new to the steering committee.

**Comment by Andersson007** [comment]
@mariolenz 
>If this doesn't make sense

I believe it does:) Good idea, thank you!
Should be very helpful and make (potential) reviewer's life much easier.
I'll add the categories.

**Comment by Andersson007** [comment]
i suggest relaxing the requirements with the following (it's not final wording, only meaning):
- One Steering committee review and approval and one non-Steering committee person's (not involved in the collection) review and approval is a requirement for the collection to be included in the package.

We could advertise this via Bullhorn among the community (in particular, among collection contributors and maintainers) as a good way:
- to contribute to the package enhancement
- to improve skills
- to get into the Steering Committee

We could announce new inclusion requests via Bullhorn regularly and it'll help:
- Make the whole community involved in the process
- Inform the community about new collections that can be included in the future

What do you think? If there's silence (i.e. no major objections), I'll create a PR and will open a vote. Meanwhile, the backlog of submitted collections is growing..

**Comment by felixfontein** [comment]
I don't think that relaxing the requirements is a good solution for this problem. I definitely would like more community interaction (which also means: reviews by the community), but I think every community review should also be checked by at least one ST person whether the review and the changes asked for makes sense. And if one review is by a non-ST person, I think it's still reasonable to expect someone from ST to take a quick look at the collection for obvious problems that were missed in the reviews.

**Comment by Andersson007** [comment]
> I don't think that relaxing the requirements is a good solution for this problem. I definitely would like more community interaction (which also means: reviews by the community), but I think every community review should also be checked by at least one ST person whether the review and the changes asked for makes sense. And if one review is by a non-ST person, I think it's still reasonable to expect someone from ST to take a quick look at the collection for obvious problems that were missed in the reviews.

Sounds sensible to me, so how about (this is a very rough text):
> At least one Steering committee review and approval and one non-Steering committee person's (not involved in the collection) review and approval (checked and approved by another SC member) is a requirement for the collection to be included in the package.

So there will be still 2 SC members involved but there will be less work for them. Thoughts?

**Comment by Andersson007** [comment]
The vote on merging the [PR](https://github.com/ansible/ansible/pull/77905) has been just opened.

As an experiment described in [another topic](https://github.com/ansible-community/community-topics/issues/104#issuecomment-1137702176), I've created the [vote dicsussion](https://github.com/ansible-community/community-topics/discussions/106).
Please vote in the [discussion](https://github.com/ansible-community/community-topics/discussions/106) (with +1/-1), NOT here.

To avoid the noise caused by voting, subscribe only for issues:
1. Click the down arrow on the `Watch/Unwatch` button
2. Choose `Custom`
3. Choose at least `Issues`

The vote will end on June 3, 2022

**Comment by mariolenz** [comment]
> 1. Click the down arrow on the `Watch/Unwatch` button
> 2. Choose `Custom`
> 3. Choose at least `Issues`
> 4. After voting in the discussion, click `Unsubscribe` in the discussion

Looks a bit complicated to me. Personally, I'd rather live with just ignoring `+1` / `-1` comments. Seems easier to me.

**Comment by briantist** [comment]
> Looks a bit complicated to me. Personally, I'd rather live with just ignoring `+1` / `-1` comments. Seems easier to me.

To be fair, all of those steps are optional. Steps 1 through 3 are only done once.

But yes, to me, this is the same as unsubscribing from an issue after voting. The only difference is that using discussions, you won't get the `+1` notifications _before_ you vote. I'm ok with using discussion comments for the vote because to me it's basically the same as the status quo, but I was looking for something more in #104 .

**Comment by Andersson007** [comment]
> > 1. Click the down arrow on the `Watch/Unwatch` button
> > 2. Choose `Custom`
> > 3. Choose at least `Issues`
> > 4. After voting in the discussion, click `Unsubscribe` in the discussion
> 
> Looks a bit complicated to me. Personally, I'd rather live with just ignoring `+1` / `-1` comments. Seems easier to me.

@mariolenz  , as @briantist mentioned, steps 1-3 need to be done only once. Benefits of this compared to ignoring votes:
1. You don't get notifications when someone votes before you vote
2. After you vote in the discussion and click unsubscribe there, you won't get further vote notifications

I.e. after you do steps 1-3 only once, the vote algorithm is simple: vote + click unsubscribe in a discussion.

I can see now that many folks already voted in the [discussion](https://github.com/ansible-community/community-topics/discussions/106) and i've got nothing, feels good:)

**Comment by Andersson007** [comment]
The vote has ended.
Results counted in the [discussion](https://github.com/ansible-community/community-topics/discussions/106):
- Community: 2 +1, nobody is against
- Steering Committee: 8 +1, nobody against

Could anyone from the Steering Committee please confirm my numbers?

**Comment by markuman** [comment]
I count the same numbers this time.

**Comment by Andersson007** [comment]
@markuman thanks for the confirmation!

Thanks everyone! Since now we can start promoting the inclusion reviews via Bullhorn and everywhere
